test/utils/k8s/job.go

```diff
@@ -175,19 +175,29 @@ func WaitForJob(ns string, kubeClient kubernetes.Interface, jobName string, time
 		case <-timeoutCh:
 			return nil, fmt.Errorf("Timeout: Job is still active: %s", jobName)
 		case <-tick:
-			job, _ := jobsClient.Get(context.TODO(), jobName, metav1.GetOptions{})
+			job, err := jobsClient.Get(context.TODO(), jobName, metav1.GetOptions{})
+
+			if err != nil {
+				return nil, err // Handle the error
+			}
 
 			if job.Status.Active > 0 {
 				fmt.Println("Job is still active")
 			} else if len(job.Status.Conditions) > 0 {
-				if job.Status.Conditions[0].Type == batchv1.JobComplete {
-					fmt.Println("Job Successful!")
-					return job, nil
-				} else if job.Status.Conditions[0].Type == batchv1.JobFailed {
-					statusJson, _ := json.Marshal(job.Status)
-					fmt.Printf("Job failed?, status = %v\n", string(statusJson))
-					return job, fmt.Errorf("Job failed. Status: %s", string(statusJson))
+				for _, condition := range job.Status.Conditions {
+					switch condition.Type {
+					case batchv1.JobComplete:
```

Comment:
Here we possibly have an issue similar to what we had in a retry function fixed recently.
The right thing to do is to go through all conditions first, before returning an error on the first
found failed condition. That is because the order is not guaranteed, so the complete might show
after a failed condition.

Reply:
@fgiorgetti  as we discussed, the Complete and Failed conditions are terminal conditions, and mutually exclusive.

https://kubernetes.io/docs/concepts/workloads/controllers/job/#terminal-job-conditions